Math/NumberTheory/ArithmeticFunctions/Standard.hs

```diff
@@ -340,40 +340,17 @@ instance Num a => Monoid (ListProduct a) where
   mempty  = ListProduct mempty
   mappend = (<>)
 
--- | Represents an upper bound on a datatype
-data Bound a = Closed a | Open 
+-- Represent as a Reader monad
+newtype BoundedSetProduct a = BoundedSetProduct { getBoundedSetProduct :: a -> Set a }
 
-instance Ord a => Semigroup (Bound a) where
-  Open      <> b         = b
-  b         <> Open      = b
-  Closed b1 <> Closed b2 = Closed (min b1 b2)
+instance (Ord a, Num a) => Semigroup (BoundedSetProduct a) where
+  BoundedSetProduct f1 <> BoundedSetProduct f2 = BoundedSetProduct f
+    where f b = s1 <> s2 <> foldMap (\n -> S.takeWhileAntitone (<=b) $ S.mapMonotonic (* n) s2) s1
```

Comment:
`S.takeWhileAntitone` is a comparably new addition to `containers`, which causes [GHC 8.0.2 build](https://travis-ci.org/github/Bodigrim/arithmoi/jobs/666447515) to fail.
Please use `S.split` or `S.splitMember` instead.